CMakeLists.txt

```diff
@@ -26,6 +26,7 @@ option(BUILD_TRANSLATIONS "" OFF)
 option(BUILD_SHARED_LIBS "" OFF)
 option(CHATTERINO_LTO "Enable LTO for all targets" OFF)
 option(CHATTERINO_PLUGINS "Enable EXPERIMENTAL plugin support in Chatterino" OFF)
+option(CHATTERINO_DISABLE_UPDATER "Disable update checks" OFF)
```

Comment:
This appears to be the only flag that disables something instead of enabling something. Should this be flipped?

Reply:
I considered this, but the cmake flag `CHATTERINO_DISABLE_UPDATER` is more closely aligned what I'd like the code in `Updates.cpp` to look like

Reply:
Seems like a matter of replacing `#ifdef` with `#ifndef`, or am I missing something?

Reply:
You're right, it's a question of

```c++
#ifdef CHATTERINO_DISABLE_UPDATER
   return;
#endif
```

which I read as "If the updater is disabled"

vs

```c++
#ifndef CHATTERINO_ENABLE_UPDATER
   return;
#endif
```

which I read as "If the updater is not enabled"

and to me, the upper option more clearly describes the intent

100% nitpicking and I would be fine with either, especially since it has such a small scope, only being used in one place in one file

Reply:
1) I believe you would usually wrap `#ifdef CHATTERINO_ENABLE_UPDATER` around the entire body of the function (after `{` and before `}`) in order to avoid warnings of unreachable code (since nothing after the return can be executed).

2) The issue I have with an outlying inverted boolean like ...DISABLE.. this is that it's not logical when you look at it in the list of options. I'd wager that the `#ifdef` is not be read as often as the name of the flag in either the source, cmake output (if that ever shows it) or potentional CI scripts. The implementation would therefore be the better place to handle this logical oddity. But if 1) was applied then there would be no negation anyways.

Reply:
If the argument is namesake is there not a world where we just call it `CHATTERINO_UPDATER` and have it ON? OFF being no update checks

Reply:
> 1. I believe you would usually wrap #ifdef CHATTERINO_ENABLE_UPDATER around the entire body of the function (after { and before }) in order to avoid warnings of unreachable code (since nothing after the return can be executed).

That's fair, the code should be updated so it's not just an early out

> 2. The issue I have with an outlying inverted boolean like ...DISABLE.. this is that it's not logical when you look at it in the list of options.

True, the option should be marked as advanced so it doesn't show up in GUIs unless the user knows what they're doing

The latest commit means the builds that don't want to use the built-in updater would write
`cmake -DCHATTERINO_UPDATER=OFF ..`

If you open it in gui like `cmake-gui` the option won't show up as to not confuse people.

If `CHATTERINO_UPDATER` is set to *OFF* or *NO* or *0* by the user compiling Chatterino, the compile definition `CHATTERINO_DISABLE_UPDATER` is added (as to not add a compile definition for 99.9% of compiles)

The full `checkForUpdates` code-block is wrapped in `#ifndef CHATTERINO_DISABLE_UPDATER` to make sure there's no unreachable code warnings